### PR TITLE
Use new FORMS_ENV variable in HostingEnvironment

### PR DESCRIPTION
**Pull request by DavidBiddle**
#### What problem does the pull request solve?
We have a class `HostingEnvironment` which allows the application to know which environment it's running in. This uses the PAAS_ENVIRONMENT variable. However, this value is not available in AWS environments. This means it will always return `unknown-environment` in AWS environments.

We use this class to determine:
- which tag should show in the header (so at present it will show a blue 'Unknown environment' tag in the header)
- whether ViewComponent should render previews in the environment (at present they won't render in the AWS dev environment)
- 
We've now added a FORMS_ENV variable to pass this information into both PaaS and AWS app instances, so this PR updates the appicaltion to use these values.

Trello card: Connected to https://trello.com/c/DFOs5YuI/858-display-which-environment-a-user-is-accessing

#### Checklist

- [x] I've used the pull request template
- [x] I've linked this PR to the relevant issue (if mission work)
- [x] I've written unit tests for these changes (if code change)
- [n/a] I've updated the documentation in (If any documentation requires updating)
  - [n/a] README.md
  - [n/a] Elsewhere (please link)

### CODE REVIEW

File: app/lib/hosting_environment.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def self.local_development?
   end
 
   def self.dev?
-    environment_name == "dev"
+    environment_name == "dev" || environment_name == "paas_dev"
```
Comment (aliuk2012) [inline app/lib/hosting_environment.rb:17]: I;m a bit confused about this, I thought we agreed not to mention platform which the app is deployed too. I really dont think our app should worry about whether its in paas or aws. it should only care about whether its dev/staging/prod/ur

Comment (DavidBiddle) [reply app/lib/hosting_environment.rb:17]: We agreed that we shouldn't put anything platform-related in the variable name, but Dan said we could put platform information in the variable values if necessary (in this case adding the `paas_` prefix enables Laurence's script to differentiate between the different platforms)

File: app/lib/hosting_environment.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module HostingEnvironment
   def self.environment_name
-    ENV.fetch("PAAS_ENVIRONMENT", "unknown-environment")
+    ENV.fetch("FORMS_ENV", "unknown-environment")
```
[the remark this answers is not in the document]
Comment (DavidBiddle) [reply app/lib/hosting_environment.rb]: Good idea, have made this change. Will also need to merge this forms-deploy PR: https://github.com/alphagov/forms-deploy/pull/227

File: config/settings.yml [hunk 3]
```diff
@@ -46,3 +46,5 @@ basic_auth:
     name: GDS User Research
     slug: gds-user-research
     govuk_content_id: "00000000-0000-0000-0000-000000000000"
+
+forms_env: local
```
[the remark this answers is not in the document]
Comment (DavidBiddle) [reply config/settings.yml:50]: Done!

**Review by aliuk2012 — APPROVED** [review]

**Pull request merged**